src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -870,7 +872,19 @@ private void ScanSyntaxToken(ref TokenInfo info)
                         TextWindow.AdvanceChar();
                     }
 
-                    info.Text = TextWindow.GetText(intern: true);
+                    if (_errorCount++ > 100)
```

Comment:
I don't object to using the same value as the native compiler used.

:question: What about the special error message if you hit the limit?